bin/recursive_gen/src/main.cpp

```diff
@@ -52,6 +53,58 @@
 #include <nil/blueprint/transpiler/public_input.hpp>
 #include <nil/blueprint/transpiler/recursive_verifier_generator.hpp>
 
+template<typename StreamType>
+std::optional<StreamType> open_file(const std::string& path, std::ios_base::openmode mode) {
+    StreamType file(path, mode);
+    if (!file.is_open()) {
+        std::cerr << "Unable to open file: " << path;
+        return std::nullopt;
+    }
+
+    return file;
+}
+
+std::optional<std::vector<std::uint8_t>> read_file_to_vector(const std::string& path) {
```

Comment:
Yeah. It's not best option.
But in fact there won't be really large data for it in production.
We won't read full assignment table here -- we will read only common_data and constraint_system.

These functions were copied from proof_producer. 
Maybe it'll be better to discuss the best implementation for all read/write functionality and place it somewhere to prevent code duplication.